capstone/src/main/webapp/youtubeScript.js

```diff
@@ -171,11 +204,83 @@ function showAvailableAttributes() {
     label.appendChild(document.createTextNode(attribute));
     avaiableAttributesElement.appendChild(checkbox);
     avaiableAttributesElement.appendChild(label);
-    avaiableAttributesElement.appendChild (document.createTextNode (" "));
+    avaiableAttributesElement.appendChild(document.createTextNode (" "));
   });
 }
 
 /** Checks if a character is a letter */
 function isLetter(character) {
   return (character.charCodeAt() >= 65 && character.charCodeAt() <= 90) || (character.charCodeAt() >= 97 && character.charCodeAt() <= 122); 
 }
+
+/** Fetches top videos based categoty Id */
+async function getTrending(categoryId) {
+  const trendingResponse = await fetch('/trending_servlet?videoCategoryId=' + categoryId);
+  const trendingResponseJson = await trendingResponse.json();
+  const trendingVideoIds = [];
+  for (const item in trendingResponseJson.items) {
+    const videoId = trendingResponseJson.items[item].id;
+    trendingVideoIds.push(videoId);
+  }
+  const commentsList = []
+  for (const id in trendingVideoIds) {
+    const videoCommentList = await fetch('/youtube_servlet?videoId=' + trendingVideoIds[id]);
+    const videoCommentListJson = await videoCommentList.json();
+    commentsList.push(videoCommentListJson);
+  }
+  inputCommentsToPerspective(commentsList);
+}
+
+/** Enables and disables input into the text field */
+function textInputToggle(button, toEnable) {
+  if (button.checked) {
+    if (toEnable) {
+      document.getElementById('channelIdForAnalysis').value = button.id;
+      document.getElementById('channelIdForAnalysis').disabled = true;
+    } else {
+      document.getElementById('channelIdForAnalysis').value = "";
+      document.getElementById('channelIdForAnalysis').disabled = false;
+    }
+  }
+}
+
+/** Creates radio buttons to allow the user to select between various categories*/
+function showCategories() {
```

Comment:
Just FYI:

This is fine for now, but just FYI another way you might be able to achieve the same functionality is by creating all these elements in your .html file under the `categoryContainer` but adding the style `display: none;`.  Then when you want to show the categories you could change it to `display: block` (or similar).  The advantage of that is it might be a little easier to read, although the disadvantage is you wouldn't be creating each category item in a loop in JavaScript.

Also frameworks like Angular/React/etc typically help in cases like this where you may want to create 1 HTML element per item in a JS array.  They let you write HTML templates which get applied to each element in the array.